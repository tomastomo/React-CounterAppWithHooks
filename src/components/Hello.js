import React from 'react';
import PropTypes from 'prop-types';

const Hello = ({welcome, subtitle}) => {
    return (
        <>
            <h1>{welcome}</h1>
            <p>{subtitle}</p>
        </>
    );
}

Hello.propTypes ={
    welcome: PropTypes.string,
}

Hello.defaultProps ={
    subtitle: 'My first React page',
}

export default Hello;