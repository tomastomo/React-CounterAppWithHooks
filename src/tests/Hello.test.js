import React from 'react';
import { render } from "@testing-library/react";
import  { shallow } from 'enzyme';
import Hello from '../components/Hello';
import '@testing-library/jest-dom';


describe('Test in <Hello>', () => {
    test('should show Hi from React', () => {
        
            const w = 'Hi From React';
            const wrapper = shallow(<Hello welcome={w}/>)

            expect(wrapper).toMatchSnapshot();
        
    });

    test('should show the subtitle', () => {
        const w = 'Hi From React';
        const s = 'Hi subtitle'

        const wrapper = shallow(
        <Hello 
            welcome={w}
            subtitle={s}
        />);
        
        const text = wrapper.find('p').text();
        expect(text).toBe(s);
    })

    
    
    
})
