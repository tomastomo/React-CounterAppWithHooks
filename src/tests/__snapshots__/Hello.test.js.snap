// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Hello> should show Hi from React 1`] = `
<Fragment>
  <h1>
    Hi From React
  </h1>
  <p>
    My first React page
  </p>
</Fragment>
`;
